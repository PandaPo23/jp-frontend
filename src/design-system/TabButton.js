import React from 'react';

import targetedProps from '../utils/targetedProps';
import Button from './Button';

const defaults = {
  borderRadius: 'top.sm',
  borderColor: 'transparent',
  borderTop: 0,
  borderBottom: 4,
  borderLeft: 0,
  borderRight: 0,
  boxShadow: 0,
  bold: false,
  uppercase: true,
  bg: 'surface',
  px: 3,
  py: 3,
  minWidth: 'auto',
  color: 'misc.muted',
  letterSpacing: 'tracked',
};

const activeDefaults = {
  borderColor: 'secondary',
  bold: true,
  color: 'on.background',
};

const TabButton = ({ active = false, children, ...props }) => {
  const activeProps = { ...activeDefaults, ...targetedProps(props, 'active') };
  return (
    <Button {...defaults} {...(active && activeProps)} {...props}>
      {children}
    </Button>
  );
};

export default TabButton;
