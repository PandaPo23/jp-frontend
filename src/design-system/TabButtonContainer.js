import system from '../utils/System';
import Flex from './Flex';

const TabButtonContainer = system(
  'TabButtonContainer',
  {
    extend: Flex,
  },
  (props) => ({
    flexWrap: 'wrap',
  })
);

export default TabButtonContainer;
