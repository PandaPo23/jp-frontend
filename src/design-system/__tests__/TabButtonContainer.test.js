import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import TabButton from '../TabButton';
import TabButtonContainer from '../TabButtonContainer';

describe('TabButtonContainer component', () => {
  const getComponent = (props) => (
    <TabButtonContainer>
      <TabButton>Button</TabButton>
    </TabButtonContainer>
  );
  it('should render TabButtonContainer', () => {
    const wrapper = mount(<TabButtonContainer />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render TabButtonContainer with children', () => {
    const component = getComponent();
    const wrapper = mount(component);
    expect(wrapper.find(TabButton).length).toBe(1);
  });
});
