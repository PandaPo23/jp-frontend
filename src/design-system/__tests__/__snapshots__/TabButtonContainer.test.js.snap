// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabButtonContainer component should render TabButtonContainer 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<TabButtonContainer
  display="flex"
>
  <div
    className="emotion-0"
  />
</TabButtonContainer>
`;
